services/roomService/src/lib/CoveyTownController.test.ts

```diff
@@ -312,3 +312,129 @@ describe('Chat Messages', () => {
+    mockSockets[playerIndex].on.mock.calls.forEach(call => {
+      if (call[0] === 'sendChatMessage')
+        call[1]({
+          message,
+          broadcastRadius,
+        });
+    });
+  }
+  
+  function blockPlayer(playerIndex: number, playerID: string) {
+    mockSockets[playerIndex].on.mock.calls.forEach(call => {
+    if (call[0] === 'blockPlayerInChat')
+      call[1](playerID);
+    });
+  }
+
+  function unblockPlayer(playerIndex: number, playerID: string) {
+    mockSockets[playerIndex].on.mock.calls.forEach(call => {
+    if (call[0] === 'unblockPlayerInChat')
+      call[1](playerID);
+    });
+  }
+
+
+  it('should broadcast message to the nearby unblocked players and not to the blocked players', async () => {
+    // Player 0 blocks player 1 and player2
+    blockPlayer(0, players[1].id);
+    blockPlayer(0, players[2].id);
+    // Player 0 sends a message via the socket
+    sendMessage(0, 'Hello World', 80);
+    // Check player 1 and player2 does not recieve the message
+    expect(mockSockets[1].emit).not.toHaveBeenCalledWith('receiveChatMessage', expect.anything());
+    expect(mockSockets[2].emit).not.toHaveBeenCalledWith('receiveChatMessage', expect.anything()); 
+    // Check other players receive the message
+    expect(mockSockets[0].emit).toHaveBeenCalledWith('receiveChatMessage', expect.anything());
+    expect(mockSockets[3].emit).toHaveBeenCalledWith('receiveChatMessage', expect.anything());
+  });
+
+  it('should broadcast message to the player again when he/she is unblocked', async () => {
+    // Player 0 blocks player 1
+    blockPlayer(0, players[1].id);
+    // Player 0 sends a message via the socket
+    sendMessage(0, 'Hello World', 80);
+    // Check player 1 does not recieve the message
+    expect(mockSockets[1].emit).not.toHaveBeenCalledWith('receiveChatMessage', expect.anything());
+    // Check other players receive the message
+    expect(mockSockets[0].emit).toHaveBeenCalledWith('receiveChatMessage', expect.anything());
+    expect(mockSockets[2].emit).toHaveBeenCalledWith('receiveChatMessage', expect.anything()); 
+    expect(mockSockets[3].emit).toHaveBeenCalledWith('receiveChatMessage', expect.anything());
+    // player 0 unblocks player 1
+    unblockPlayer(0, players[1].id);
+    // Player 0 sends a message via the socket
+    sendMessage(0, 'Hello World Again', 80);
+    // check if all players including player 1 receive the message
+    mockSockets.forEach(socket => expect(socket.emit).toHaveBeenCalledWith('receiveChatMessage', expect.anything()));
+  });
+
+  it('player should receive message as long as sender is not the player who blocked him/her', async () => {
+    // player 0 and player 1 blocks player 3
+    blockPlayer(0, players[3].id);
+    blockPlayer(1, players[3].id);
+    
+    sendMessage(0, 'weather is hot today', 80);
+    sendMessage(1, 'weather is cold today', 80);
+    sendMessage(2, 'weather is warm today', 80);
+    
```

Comment:
You should add another expect here
`expect(mockSockets[3].emit).toHaveBeenCalledTimes(1);`

Without this, this test will pass even if the two blockPlayer calls are removed